File: app/src/main/java/cn/alvkeke/dropto/ui/intf/NoteAttemptListener.java
package cn.alvkeke.dropto.ui.intf;

import java.util.ArrayList;

import cn.alvkeke.dropto.data.NoteItem;

public interface NoteAttemptListener {
    enum Attempt {
        CREATE,
        REMOVE,
        UPDATE,
        COPY,
        SHOW_DETAIL,
        SHOW_SHARE,
        SHOW_IMAGE,
    }
    void onAttempt(Attempt attempt, NoteItem e);
    void onAttempt(Attempt attempt, NoteItem e, Object ext);
    void onAttemptBatch(Attempt attempt, ArrayList<NoteItem> noteItems);
}

File: app/src/main/java/cn/alvkeke/dropto/ui/intf/NoteAttemptListener.kt
package cn.alvkeke.dropto.ui.intf

import cn.alvkeke.dropto.data.NoteItem

interface NoteAttemptListener {
    enum class Attempt {
        CREATE,
        REMOVE,
        UPDATE,
        COPY,
        SHOW_DETAIL,
        SHOW_SHARE,
        SHOW_IMAGE,
        SHOW_FORWARD,
    }

    fun onAttempt(attempt: Attempt, e: NoteItem)
    fun onAttempt(attempt: Attempt, e: NoteItem, ext: Any?)
    fun onAttemptBatch(attempt: Attempt, noteItems: ArrayList<NoteItem>)
}
